pkg/plugin/processor/builtin/webhook_http.go

```diff
@@ -0,0 +1,365 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"maps"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/goccy/go-json"
+	"github.com/jpillora/backoff"
+	"github.com/mitchellh/mapstructure"
+)
+
+var defaultWebhookHTTPConfig = map[string]string{
+	"request.method":      http.MethodPost,
+	"request.contentType": "application/json",
+
+	"backoffRetry.count":  "0",
+	"backoffRetry.factor": "2",
+	"backoffRetry.min":    "100ms",
+	"backoffRetry.max":    "5s",
+
+	"request.body":  ".",
+	"response.body": ".Payload.After",
+}
+
+type webhookHTTPConfig struct {
+	// URL used in the HTTP request.
+	URL string `mapstructure:"request.url"`
+	// HTTP request method to be used.
+	Method string `mapstructure:"request.method"`
+	// Value of the Content-Type header.
+	ContentType string `mapstructure:"request.contentType"`
+
+	// Maximum number of retries for an individual record when backing off following an error.
+	// Defaults to 0.
+	BackoffRetryCount float64 `mapstructure:"backoffRetry.count"`
+	// The multiplying factor for each increment step.
+	// Default: 2.
+	BackoffRetryFactor float64 `mapstructure:"backoffRetry.factor"`
+	// Minimum waiting time before retrying.
+	// Default: 100ms
+	BackoffRetryMin time.Duration `mapstructure:"backoffRetry.min"`
+	// Maximum waiting time before retrying.
+	// Default: 5s
+	BackoffRetryMax time.Duration `mapstructure:"backoffRetry.max"`
+
+	// RequestBodyRef specifies which field from the input record
+	// should be used as the body in the HTTP request.
+	// The value of this parameter should be a valid record field reference:
+	// See: sdk.NewReferenceResolver
+	RequestBodyRef *sdk.ReferenceResolver `mapstructure:"request.body"`
+	// ResponseBodyRef specifies to which field should the
+	// response body be saved to.
+	// The value of this parameter should be a valid record field reference:
+	// See: sdk.NewReferenceResolver
+	ResponseBodyRef *sdk.ReferenceResolver `mapstructure:"response.body"`
+	// ResponseStatusRef specifies to which field should the
+	// response status be saved to.
+	// The value of this parameter should be a valid record field reference:
+	// See: sdk.NewReferenceResolver
+	ResponseStatusRef *sdk.ReferenceResolver `mapstructure:"response.status"`
+}
+
+type webhookHTTP struct {
+	sdk.UnimplementedProcessor
+
+	logger log.CtxLogger
+	config webhookHTTPConfig
+}
+
+func NewWebhookHTTP(l log.CtxLogger) sdk.Processor {
+	return &webhookHTTP{logger: l.WithComponent("builtin.webhookHTTP")}
```

Comment:
Hmm, not sure exactly why we provided the logger here, since the logger will also be provided through the context. Generally the idea is to use use `sdk.Logger(ctx)` to fetch the logger.

No need to change this now, it's a small thing that can be fixed later on, but we need to make sure we decide on a way to handle logging here.

Reply:
Agreed! Yeah, I wanted to make it compatible with the ProcessorPluginConstructor.